File: queue/0933-number-of-recent-calls/cpp/Solution.cpp
#include <queue>
using namespace std;

class RecentCounter {
private:
    queue<int> recentCalls;

public:
    RecentCounter() = default;

    int ping(int t) {
        recentCalls.push(t);

        while (!recentCalls.empty() && recentCalls.front() < t - 3000) {
            recentCalls.pop();
        }

        return static_cast<int>(recentCalls.size());
    }
};

File: queue/0933-number-of-recent-calls/java/Solution.java
import java.util.ArrayDeque;
import java.util.Queue;

class RecentCounter {
    private final Queue<Integer> recentCalls;

    public RecentCounter() {
        recentCalls = new ArrayDeque<>();
    }

    public int ping(int t) {
        recentCalls.offer(t);

        while (!recentCalls.isEmpty() && recentCalls.peek() < t - 3000) {
            recentCalls.poll();
        }

        return recentCalls.size();
    }
}
